packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -491,16 +502,27 @@ def __init__(
         """
         AbciAppDB._check_data(setup_data)
         self._setup_data = deepcopy(setup_data)
-        self._cross_period_persisted_keys: Set[str] = (
-            cross_period_persisted_keys.copy()
-            if cross_period_persisted_keys is not None
-            else set()
-        )
         self._data: Dict[int, Dict[str, List[Any]]] = {
             RESET_COUNT_START: self.setup_data  # the key represents the reset index
         }
         self._round_count = ROUND_COUNT_DEFAULT  # ensures first round is indexed at 0!
 
+        self._cross_period_persisted_keys = self.default_cross_period_keys.union(
```

Comment:
the resulting will be set not frozenset afaik

Reply:
```
assert isinstance(
    frozenset().union({"test"}), frozenset
), "`frozenset().union` should produce a `frozenset`."
```